File: app/src/main/java/me/gabeg/sicksends/SsSportDao.java
package me.gabeg.sicksends;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import java.util.List;

/**
 * Data access object for sport climbing.
 */
@Dao
public interface SsSportDao
	extends SsProblemDao<SsSport>
{

	/**
	 * @return All sport problems.
	 */
	@Query("SELECT * FROM sport")
	LiveData<List<SsSport>> getAll();

}

File: app/src/main/java/me/gabeg/sicksends/db/sport/SsSportDao.kt
package me.gabeg.sicksends.db.sport

import androidx.lifecycle.LiveData
import androidx.room.Dao
import me.gabeg.sicksends.db.generic.SsGenericProblemDao
import androidx.room.Query
import me.gabeg.sicksends.db.generic.SsGenericProblem

/**
 * Data access object for sport climbing.
 */
@Dao
interface SsSportDao : SsGenericProblemDao<SsSportProblem>
{

	/**
	 * Delete all sport problems.
	 */
	//@Query("DELETE FROM sport")
	//override suspend fun deleteAll()

	/**
	 * @see SsGenericProblemDao.findProblems
	 */
	@Query("SELECT * FROM sport ${SsGenericProblem.WHERE} ${SsGenericProblem.ORDER_BY}")
	override fun findProblems(
		isOutdoor : Boolean?,
		isProject : Boolean?,
		isFlash : Boolean?) : LiveData<List<SsSportProblem>>

	/**
	 * Get all sport problems.
	 *
	 * @return All sport problems.
	 */
	@Query("SELECT * FROM sport ${SsGenericProblem.ORDER_BY}")
	override fun getAllProblems() : LiveData<List<SsSportProblem>>

}
